src/arrayMethodReduce.js

```diff
@@ -5,7 +5,27 @@
  */
 function applyCustomReduce() {
   [].__proto__.reduce2 = function(callback, initialValue) {
-    // write code here
+    if (this.length === 0) {
+      return initialValue;
+    }
+
+    let acc;
+
+    if (initialValue !== undefined) {
+      acc = initialValue;
+
+      for (let i = 0; i < this.length; i++) {
+        acc = callback(acc, this[i], i, this, initialValue);
+      }
+    } else {
+      acc = this[0];
+
+      for (let i = 1; i < this.length; i++) {
+        acc = callback(acc, this[i], i, this, initialValue);
+      }
+    }
```

Comment:
remove this if else, just rewrite the acc variable on condition
also write loop only once